sacred/observers/base.py

```diff
@@ -37,6 +37,12 @@ def resource_event(self, filename):
     def artifact_event(self, name, filename, metadata=None, content_type=None):
         pass
 
+    def log_metrics(self, metrics_by_name, info):
+        pass
+
+    def join(self):
```

Comment:
Do we need join in the common interface? I don't think it is called from run anywhere.

Reply:
Join is called here: https://github.com/IDSIA/sacred/blob/9272118f30362806ec7f45f7c1e7804d4baf6427/sacred/run.py#L442 So I think we need it too.

Reply:
Ah, I see.